test/tests/functional/pbs_basil_support.py

```diff
@@ -362,5 +407,393 @@ def test_hbmemm_rsc(self):
+
+        # Check that job is in Q state with valid comment
+        job_comment = "Not Running: Insufficient amount of resource: hbmem"
+        self.server.expect(JOB, {'job_state': 'Q', 'comment':
+                           (MATCH_RE, job_comment)}, id=job_id)
+
         # Delete the Job.
         self.server.delete(job_id, wait=True)
+
+    def test_job_request_knl(self):
+        """
+        Create a job that requests aoe should run on a KNL vnode.
+        Submit the job to the Server.Check if the job runs on a KNL vnode
+        and if the job is in the 'R' state.
+        """
+        if self.du.platform == 'craysim':
+            self.skipTest(reason='Test is NA for Craysim')
+
+        # Find the list of KNL vnodes
+        klist = self.retklist()
+        if len(klist) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (klist))
+
+        # Set provisioning attributes on KNL vnode.
+        self.set_provisioning()
+
+        # Submit job that request aoe
+        req_aoe = self.request_current_aoe()
+        job = Job(TEST_USER)
+
+        job.create_script(
+                          "#PBS -joe -o localhost:/tmp -lselect=1:ncpus=1:aoe=%s\n" % req_aoe +
+                          "cd /tmp\n" +
+                          "aprun -B sleep 10\n" +
+                          "sleep 10")
+
+        job_id = self.server.submit(job)
+        self.server.expect(JOB, {'job_state': 'R'}, id=job_id)
+
+        # Check that exec_vnode is a KNL vnode.
+        self.server.status(JOB, 'exec_vnode', id=job_id)
+        evnode = job.execvnode()[0].keys()[0]
+        if evnode in klist:
+            self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
+            rv = 1
+        else:
+            self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
+            rv = 0
+        self.assertTrue(rv == 1)
+
+        # Delete the Job.
+        self.server.delete(job_id, wait=True)
+
+        # Unset provisioning attributes.
+        self.unset_provisioning()
+
+    def test_job_request_subchunk(self):
+        """
+        Test job request consist of subchunks with and without aoe resource.
+        """
+        if self.du.platform == 'craysim':
+            self.skipTest(reason='Test is NA for craysim')
+
+        # Find the list of KNL vnodes
+        klist = self.retklist()
+        if len(klist) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (klist))
+
+        # Set provisioning attributes.
+        self.set_provisioning()
+
+        # Submit job that request sub-chunk with and without aoe resources
+        req_aoe = self.request_current_aoe()
+        job = Job(TEST_USER)
+
+        job.create_script(
+                "#PBS -joe -o localhost:/tmp -lselect=1:ncpus=1:aoe=%s+1:ncpus=12\n" % req_aoe +
```

Comment:
You should use place=scatter.  If not, it's possible that the second chunk could be placed on the same vnode as the first chunk.  Once you use scatter, use ncpus=1 so the test doesn't fail if the second vnode has less than 12 cpus.

Reply:
Implemented Scatter , as suggested